symengine/infinity.cpp

```diff
@@ -143,15 +144,91 @@ RCP<const Number> Infty::div(const Number &other) const
             return infty(this->_direction->mul(*minus_one));
     }
 }
-// TODO
+
 RCP<const Number> Infty::pow(const Number &other) const
 {
-    return zero;
+    if (is_a<Infty>(other)) {
+        if (is_positive_infinity()) {
+            if (other.is_negative()) {
+                return zero;
+            } else if (other.is_positive()) {
+                return rcp_from_this_cast<Number>();
+            } else {
+                throw SymEngineException("Indeterminate Expression: `Infty ** "
+                                         "unsigned Infty` encountered");
+            }
+        } else if (is_negative_infinity()) {
+            throw NotImplementedError("NaN module not yet implemented");
+        } else {
+            if (other.is_positive()) {
+                return infty(0);
+            } else if (other.is_negative()) {
+                return zero;
+            } else {
+                throw SymEngineException("Indeterminate Expression: `unsigned "
+                                         "Infty ** unsigned Infty` "
+                                         "encountered");
+            }
+        }
+    } else if (is_a<Complex>(other)) {
+        throw NotImplementedError(
+            "Raising to the Complex powers not yet implemented");
+    } else {
+        if (other.is_negative()) {
+            return zero;
+        } else if (other.is_zero()) {
+            return one;
+        } else {
+            if (is_positive_infinity()) {
+                return rcp_from_this_cast<Number>();
+            } else if (is_negative_infinity()) {
+                throw NotImplementedError("Raising Negative Infty to the "
+                                          "Positive Real powers not yet "
+                                          "implemented");
+            } else {
+                return infty(0);
+            }
+        }
+    }
 }
-// TODO
+
 RCP<const Number> Infty::rpow(const Number &other) const
 {
-    return zero;
+    if (is_a<Infty>(other)) {
+        return other.pow(*this);
+    } else if (is_a<Complex>(other)) {
+        throw NotImplementedError(
+            "Raising Complex powers to Infty not yet implemented");
+    } else {
+        if (other.is_negative()) {
+            throw NotImplementedError("NaN module not yet implemented");
+        } else if (other.is_zero()) {
+            throw SymEngineException("Indeterminate Expression: `0 ** +- "
+                                     "unsigned Infty` encountered");
+        } else {
+            const RealDouble &s = static_cast<const RealDouble &>(other);
```

Comment:
Actually I wanted to down-cast the value held by the object into a double and compare with 1. But using 'down_cast()' is raising an error.